testing/python/amd/test_tilelang_hip_codegen.py

```
@@ -0,0 +1,757 @@
+    pipelined_rowsum(num_stages)(x, out)
+    torch.cuda.synchronize()
+    torch.testing.assert_close(out, torch.full((M,), float(K), device="cuda"), atol=1e-4, rtol=0)
+
+
+@tilelang.testing.requires_rocm
+@pytest.mark.parametrize("num_stages", [2, 3])
+def test_pipelined_multi_stage_fp16_gemm(num_stages):
+    """
+    FP16 GEMM with T.Pipelined(num_stages>1) must launch and produce correct
+    results on ROCM — the most common pattern that triggered the LDS overflow
+    (A_s bM×bK + B_s bK×bN doubled per pipeline stage).
```

Ruff (0.15.11)

[warning] Docstring contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF002)